### PR TITLE
Ajitarasu/mob 117 as a mobile user, i want to work with clients that don't have mambu default address

### PR DESCRIPTION
**Pull request by ajitarasu**

### CODE REVIEW

File: mambuAndroid/src/main/java/com/mambu/android/CreateCustomer.java [hunk 1]
```diff
@@ -2386,12 +2398,13 @@ private void addTabs() {
...
         // tab 2 from XML
-        String contactsLabel = Contact;
-        if (isForClient) {
-            //  Starting with 4.1 all contact numbers are displayed as core fields.  Only the address remains in this tab
-            contactsLabel = res.getString(R.string.client_address);
+        if (isCustomerUsingDefaultAddress()) {
+            tabHost.addTab(tabHost.newTabSpec(contact).setIndicator(getContactLabel()).setContent(R.id.tabCreateContactFields));
+        } else {
+            LinearLayout ln = (LinearLayout) this.findViewById(R.id.tabCreateContactFields);
+            ln.setVisibility(View.GONE);
```
Comment (mdanilkis) [inline mambuAndroid/src/main/java/com/mambu/android/CreateCustomer.java]: Would this work for Groups too? For groups we also display phones in the same tab where address is.

File: mambuAndroid/src/main/java/com/mambu/android/CreateCustomer.java [hunk 2]
```diff
@@ -2426,6 +2439,26 @@ private void addTabs() {
     }
 
     /**
+     * Returns the title of the tab which contains the address
+     *
+     * @return address tab title
+     */
+    private String getContactLabel() {
+        return isForClient ? res.getString(R.string.client_address) : contact;
+    }
```
Comment (mdanilkis) [inline mambuAndroid/src/main/java/com/mambu/android/CreateCustomer.java]: Good refactoring. 
 In the javadoc part I would preserve the original comment explaining the reason for why we return different labels for the same tab, And one day phones might be moved to be core fields for groups too and then the same label again could be used.

File: mambuAndroid/src/main/java/com/mambu/android/CreateCustomer.java [hunk 3]
```diff
@@ -2426,6 +2439,26 @@ private void addTabs() {
...
+    /**
+     * Checks if current customer has Mambu default address enabled
+     *
+     * @return true it has Mambu default address enabled
+     * false otherwise
+     */
+    private boolean isCustomerUsingDefaultAddress() {
+        return mEntityRole != null
+                && mEntityRole.isUseDefaultAddress();
+    }
```
Comment (mdanilkis) [inline mambuAndroid/src/main/java/com/mambu/android/CreateCustomer.java]: Nice

File: mambuAndroid/src/main/java/com/mambu/android/CreateCustomer.java (hunk 2 above)
Comment (ajitarasu) [inline mambuAndroid/src/main/java/com/mambu/android/CreateCustomer.java]: Update javadoc.

File: mambuAndroid/src/main/java/com/mambu/android/CreateCustomer.java (hunk 1 above)
Comment (ajitarasu) [inline mambuAndroid/src/main/java/com/mambu/android/CreateCustomer.java]: Refactored for groups.

**Pull request merged**